xgitguard/file-scanner/secret_detection.py

```diff
@@ -0,0 +1,907 @@
+
+    logger.debug(f"Current formatted secrets_data_list count: {len(secrets_data_list)}")
+    # logger.debug(f"secrets_data_list: {secrets_data_list}")
+    return secrets_data_list
+
+
+def process_file_content(
+    file_path,
+    code_content,
+    keyword,
+    extension,
+    secrets_data=[],
+    is_secondary_credential=False,
+):
+    """
+    Process the Search
+        Remove unnecessary data from code content
+        Extract secret values using regex
+        Format the secrets detected
+        Return the secrets detected
+
+    params: file_path - string
+    params: code_content - string
+    params: keyword - string
+    params: extension - string
+    params: secrets_data - list
+    params: is_secondary_credential - bool
+    returns: secrets_data_list - list - Detected secrets data
+    """
+    logger.debug("<<<< 'Current Executing Function' >>>>")
+    secrets_data_list = []
+    lines = code_content.split("\n")
+    if lines:
+        keyword_count = code_content.lower().count(keyword.lower())
+
+        if len(secrets_data) >= 1 and len(secrets_data) <= 100:
+            if len(secrets_data) >= 40:
+                secrets_data = secrets_data[:40]
+            secret_data_list = format_detection(
+                keyword,
+                file_path,
+                code_content,
+                secrets_data,
+                keyword_count,
+                is_secondary_credential,
+            )
+            if secret_data_list:
+                for secret_data in secret_data_list:
+                    secrets_data_list.append(secret_data)
+        else:
+            logger.debug(
+                f"Skipping secrets_data as length is not between 1 to 20. Length: {len(secrets_data)}"
+            )
+    else:
+        logger.debug(
+            f"Skiping processing code content is empty. Content length: {len(lines)}"
+        )
+
+    return secrets_data_list
+
+
+def run_detection(
+    file_path,
+    secondary_keyword,
+    secondary_credentials,
+    keyword,
+    extension,
+    ml_prediction=True,
+    code_content="",
+):
+    """
+    Run search with Secondary Keyword and extension combination in the given file
+    Steps:
+        Get the code content for the given file
+        Clean the code content and extract secrets
+        Detect the secrets using RegEx and format secret records
+        Predict the secret data using ML model
+        Write the cleaned and detected secret data
+
+    params: file_path - string
+    params: secondary_keyword - string
+    params: extension - string
+    params: ml_prediction - Boolean - optional - Default: True
+    params: code_content - string
+    params: secrets_data - list
+    returns: None
+    """
+    logger.debug("<<<< 'Current Executing Function' >>>>")
+    # Read and Setup Global Configuration Data to reference in all process
+    try:
+        global configs
+        if configs:
+            pass
+    except:
```

Comment:
Mention excpetion name as it will handles all excpetions

Reply:
Added exception handling